pkg/glusterblock/nodeserver.go

```diff
@@ -0,0 +1,172 @@
+
+	// TODO fs should be argument
+	err = utils.MountVolume(srcPath, targetPath, "xfs", mo)
+	if err != nil {
+		return nil, err
+	}
+
+	return &csi.NodePublishVolumeResponse{}, nil
+}
+
+// NodeGetVolumeStats returns volume capacity statistics available for the
+// volume
+func (ns *NodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
+
+	// TODO need to implement volume status call
+	return nil, status.Error(codes.Unimplemented, "")
+}
+
+func (ns *NodeServer) validateNodePublishVolumeReq(req *csi.NodePublishVolumeRequest) error {
+	if req == nil {
+		return status.Error(codes.InvalidArgument, "request cannot be empty")
+	}
+
+	if req.GetVolumeId() == "" {
+		return status.Error(codes.InvalidArgument, "NodePublishVolume Volume ID must be provided")
+	}
+
+	if req.GetTargetPath() == "" {
+		return status.Error(codes.InvalidArgument, "NodePublishVolume Target Path cannot be empty")
+	}
+
+	if req.GetVolumeCapability() == nil {
+		return status.Error(codes.InvalidArgument, "NodePublishVolume Volume Capability must be provided")
+	}
+	return nil
+}
+
+// NodeUnpublishVolume unmounts the volume from the target path
+func (ns *NodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublishVolumeRequest) (*csi.NodeUnpublishVolumeResponse, error) {
+	glog.V(2).Infof("received node unpublish volume request %+v", req)
+
+	if err := ns.validateNodeUnpublishVolumeReq(req); err != nil {
+		return nil, err
+	}
+
+	targetPath := req.GetTargetPath()
+	notMnt, err := mounter.IsLikelyNotMountPoint(targetPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, status.Error(codes.NotFound, "targetpath not found")
+		}
+		return nil, status.Error(codes.Internal, err.Error())
+
+	}
+
+	if notMnt {
+		return nil, status.Error(codes.NotFound, "volume not mounted")
+	}
+
+	err = util.UnmountPath(targetPath, mounter)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &csi.NodeUnpublishVolumeResponse{}, nil
+}
+
+func (ns *NodeServer) validateNodeUnpublishVolumeReq(req *csi.NodeUnpublishVolumeRequest) error {
+	if req == nil {
+		return status.Errorf(codes.InvalidArgument, "request cannot be empty")
+	}
+
+	if req.VolumeId == "" {
+		return status.Error(codes.InvalidArgument, "NodeUnpublishVolume Volume ID must be provided")
+	}
+
+	if req.TargetPath == "" {
+		return status.Error(codes.InvalidArgument, "NodeUnpublishVolume Target Path must be provided")
+	}
+	return nil
+}
+
+// NodeGetInfo info
+func (ns *NodeServer) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
+	return &csi.NodeGetInfoResponse{
+		NodeId: ns.GfDriver.NodeID,
+	}, nil
+}
+
+// NodeGetCapabilities returns the supported capabilities of the node server
+func (ns *NodeServer) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetCapabilitiesRequest) (*csi.NodeGetCapabilitiesResponse, error) {
```

Comment:
this need to be updated check https://github.com/gluster/gluster-csi-driver/pull/149